docs/core-concepts/app-types.md

```diff
@@ -0,0 +1,125 @@
+# Types of portability in .NET Core
+
+## Introduction
+There are several ways to think about the "types" of applications you can build. Usually, the types describe 
+a certain execution model or are based on what the application can "do"; examples of these are "console application", 
+"web application" etc. All of these types of applications (and more) can be created with .NET Core, since it is 
+a general purpose development platform.
+
+However, given its unique and cross-platform nature, .NET Core also has another angle through which to observe the type of the application 
+and that is application's *portability*. Portability here means, essentially, where can you run your application and what are the 
+prerequisites that you need to satisfy in order for your application to run on a given machine. This document 
+deals with this angle, portability, and outlines the two main types of portability that .NET Core enables. 
+
+There are two main types that we can observe: 
+
+1. Portable application
+    * As a sub-category of this we have the portable application with native dependencies
+2. Self-contained application
+
+## Portable application
+Portable applications are the default type in .NET Core. They require .NET Core to be installed on the targeted machine 
```

Comment:
Nit: I think the beginning of the sentence should match the subsection title like this:
"Portable application is the default type in .NET Core...."
Or you should change the titles to be in plural form, which might be the better choice here.